### PR TITLE
Add Convex tests for listening sessions state machine

### LINKED ISSUE #157: Add Convex tests for listening sessions state machine
## Problem

`convex/listeningSessions.ts` owns the listening session lifecycle (create -> markTranscribing -> markSynthesizing -> complete/fail) and writes raw + synthesized notes. Today we rely on manual QA + API route tests, but the core state machine + note creation invariants are untested.

## Goal

Add automated Convex-level tests that make regressions in session state transitions + note writes hard to ship.

## Proposed Approach

- Adopt a Convex test harness (see `convex/README.md` references to `convex-test`).
- Add a focused test suite for listening sessions.

## Acceptance Criteria

- Test `create` inserts a session with `recording` + sane cap/warning normalization.
- Test `markTranscribing` only allowed from `recording` and patches audioUrl/duration/capReached/transcriptLive.
- Test `markSynthesizing` only allowed from `transcribing`.
- Test `complete`:
  - Rejects empty transcript.
  - Rejects invalid starting states.
  - Creates (or updates) raw transcript note.
  - Creates synthesized note + quote notes, respecting caps/dedup rules.
- Test `fail` marks session failed and clamps error length.
- Tests run in CI via `bun run test`.

### PR DESCRIPTION
**Pull request by phrazzld**
## Summary
- Extracted listening session mutation logic into exported handlers in `convex/listeningSessions.ts` to make the state machine directly testable.
- Added a focused `Vitest` suite for transition and completion behavior in `__tests__/convex/listeningSessions.test.ts`.

## Why
The listening session lifecycle (`create -> markTranscribing -> markSynthesizing -> complete/fail`) was not explicitly covered by unit tests, so regressions in transition guards and note synthesis behavior could slip through.

## What changed
- Added exported handlers:
  - `createListeningSessionHandler`
  - `markTranscribingHandler`
  - `markSynthesizingHandler`
  - `completeListeningSessionHandler`
  - `failListeningSessionHandler`
- Wired existing mutations to call these handlers.
- Added tests for:
  - `create` normalization
  - valid/invalid `markTranscribing` transitions
  - valid/invalid `markSynthesizing` transitions
  - `complete` transcript validation and state checks
  - raw transcript note create/update
  - synthesized note cap + quote dedupe behavior
  - `fail` error truncation

Closes #157

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive tests for the full listening-session lifecycle: creation, transcribing, synthesizing, completion, failure, transcript handling, raw note updates, and DB mutation validation.

* **New Features**
  * More reliable transcript handling, truncation safeguards, cap/dedup logic, and automatic generation/updating of synthesized and raw notes on completion.

* **Bug Fixes**
  * Improved ownership checks, clearer error reporting, and stricter invalid-transition handling.

* **Refactor**
  * Centralized lifecycle handlers for clearer, more robust state transitions.

## Before / After

Before:
- Listening session lifecycle/state machine not directly covered by handler-level unit tests.
- `completeListeningSessionHandler` could patch an existing `rawNoteId` without validating note ownership/book invariant.
- `failListeningSessionHandler` allowed invalid transitions (e.g. `complete -> failed`).
- Test harness still had some missing negative/regression cases.

After:
- Handler-level Vitest coverage for transitions + completion edge cases.
- `rawNoteId` update validates note belongs to the same user + same book (else `ConvexError`).
- `failListeningSessionHandler` enforces `ALLOWED_TRANSITIONS` via `assertTransition`.
- `completeListeningSessionHandler` uses `assertTransition` for the completion transition guard.
- Added focused regression tests for these cases.

### CODE REVIEW

File: __tests__/convex/listeningSessions.test.ts [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+describe("listening session state machine handlers", () => {
+  beforeEach(() => {
+    vi.spyOn(authModule, "requireAuth").mockResolvedValue(userId);
+  });
+
+  afterEach(() => {
+    vi.restoreAllMocks();
+    vi.useRealTimers();
+  });
+
```
[the remark this answers is not in the document]
Comment (phrazzld) [reply __tests__/convex/listeningSessions.test.ts]: Added ownership-denied coverage + other missing edge cases (defaults, truncation, dup assert).

**Comment by phrazzld** [comment]
## PR Unblocked

**Conflicts**: none
**CI**: green
**Reviews/Comments**: addressed remaining threads (fail transition validation; rawNote bookId invariant)

Ready for re-review.
